<a href="https://colab.research.google.com/github/111Vidya/AI-Fundametals/blob/main/AI_Jun4_RNN_LSTM_model.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# LSTM for sequence classification in the IMDB dataset
from keras.datasets import imdb
from keras.models import Sequential # sequence model
from keras.layers import Dense # hidden layers
from keras.layers import LSTM # LSTM model
from keras.layers import Embedding
from keras.preprocessing import sequence
from keras.utils import pad_sequences
# load the dataset but only keep the top n words, zero the rest. Senti Analysis
top_words = 5000 # only 500 top word from the dataset
(X_train, y_train), (X_test, y_test) = imdb.load_data(num_words=top_words)
# truncate and pad input sequences
max_review_length = 500 # length restricting to 500 words
X_train = pad_sequences(X_train, maxlen=max_review_length) # pad to understand better and every corner of the reviews
X_test = pad_sequences(X_test, maxlen=max_review_length)
# create the model
embedding_vecor_length = 32  #batch size
model = Sequential()
model.add(Embedding(top_words, embedding_vecor_length, input_length=max_review_length)) #3 defined layers
model.add(LSTM(100)) # 100 cells
model.add(Dense(1, activation='sigmoid')) #output layer 1, sigmoid -Binary classification
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy']) #define metrics/model performance
model.summary()
model.fit(X_train, y_train, epochs=1, batch_size=64)
# Final evaluation of the model
scores = model.evaluate(X_test, y_test, verbose=0)
print("Accuracy: %.2f%%" % (scores[1]*100))

17464789/17464789 [==============================] - 0s 0us/step
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 embedding (Embedding)       (None, 500, 32)           160000    
                                                                 
 lstm (LSTM)                 (None, 100)               53200     
                                                                 
 dense (Dense)               (None, 1)                 101       
                                                                 
Total params: 213301 (833.21 KB)
Trainable params: 213301 (833.21 KB)
Non-trainable params: 0 (0.00 Byte)
_________________________________________________________________
391/391 [==============================] - 282s 711ms/step - loss: 0.4841 - accuracy: 0.7650
Accuracy: 83.21%


Acc 83.21%
Embedding vector length - to create a vector(numbers) for words in the review to clasiffy to positive or negetive and truncate the length. Pass the embedded layer thru NN(LSTM). 100 LSTM cells will analyse the predictions that will be consolidated.